spec/system/post_index_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.feature 'User Post Index', type: :feature do
+  before do
+    @user = FactoryBot.create(:user)
+    @posts = FactoryBot.create_list(:post, 10, author: @user)
+    visit user_posts_path(@user)
+  end
+
+  scenario 'displays post index page content' do
+    expect(page).to have_css('.user-container')
+    expect(page).to have_css('.img-container img')
+    expect(page).to have_css('.user-name', text: @user.name)
+    expect(page).to have_css('.user-post', text: "Number of posts: #{@user.posts_counter}")
+
+    @posts.each do |post|
+      expect(page).to have_css('.post', text: post.title)
+      expect(page).to have_css('.post-content p', text: post.text)
+      expect(page).to have_css('.interactions', text: "comments: #{post.comments_counter}")
+      expect(page).to have_css('.interactions', text: "likes: #{post.likes_counter}")
+
+      within(".post[data-post-id='#{post.id}']") do
```

Comment:
- [ ] The below tests are failing, Please make sure all tests should pass without any errors.

![image](https://github.com/anyars-encarta/encarta-blog-app/assets/83803414/2abc3960-e727-4d95-be39-ca8229bcae5b)